Author: Reinhard Tartler <siretart@tauware.de>
Description: Detect that the binary is actually mplayer2
Bug-Debian: http://bugs.debian.org/699864

smplayer can handle mplayer2 if it detects its presence. Upstream's
solution to rename mplayer to mplayer2 is not acceptable for
Debian. Since executing the mplayer program to parse its version
identification output is too heavy-weight, this change introduces a
dpkg-specific, but light-weight check.

--- a/src/core.cpp
+++ b/src/core.cpp
@@ -28,6 +28,7 @@
 #endif
 
 #include <cmath>
+#include <unistd.h>
 
 #include "mplayerwindow.h"
 #include "desktopinfo.h"
@@ -1419,9 +1420,10 @@ void Core::startMplayer( QString file, d
         mplayer_bin = fi.absoluteFilePath();
 	}
 
-	if (fi.baseName().toLower() == "mplayer2") {
+	// debian/ubuntu specific check if we are using mplayer2
+	if ((fi.baseName().toLower() == "mplayer2") || !access("/usr/share/doc/mplayer2/copyright", F_OK)) {
+		qDebug("Core::startMplayer: this seems mplayer2");
 		if (!pref->mplayer_is_mplayer2) {
-			qDebug("Core::startMplayer: this seems mplayer2");
 			pref->mplayer_is_mplayer2 = true;
 		}
 	}
